jni_bindings/src/lib.rs

```diff
@@ -457,273 +536,268 @@ pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node
-        let mut patches = crate::diff::diff(doc, other);
-        if patches.is_empty() {
-            return;
-        }
-        let mut editor = doc.edit();
-        let mut stack = vec![];
-        for patch in patches.drain(..) {
-            let patch_result = patch.apply(&mut editor, &mut stack);
-            match patch_result {
-                None => (),
-                Some(diff::PatchResult::Add { node, parent }) => {
-                    env.call_method(
-                        &interface,
-                        "mOnHandle",
-                        "(JBII)V",
-                        &[
-                            JValue::Long(this),
-                            JValue::Byte(ChangeType::Add as jbyte),
-                            JValue::Int(node.as_u32() as jint),
-                            JValue::Int(parent.as_u32() as jint),
-                        ],
-                    )
-                    .unwrap();
-                }
-                Some(diff::PatchResult::Remove { node, parent }) => {
-                    env.call_method(
-                        &interface,
-                        "mOnHandle",
-                        "(JBII)V",
-                        &[
-                            JValue::Long(this),
-                            JValue::Byte(ChangeType::Remove as jbyte),
-                            JValue::Int(node.as_u32() as jint),
-                            JValue::Int(parent.as_u32() as jint),
-                        ],
-                    )
-                    .unwrap();
-                }
-                Some(diff::PatchResult::Change { node }) => {
-                    env.call_method(
-                        &interface,
-                        "mOnHandle",
-                        "(JBII)V",
-                        &[
-                            JValue::Long(this),
-                            JValue::Byte(ChangeType::Change as jbyte),
-                            JValue::Int(node.as_u32() as jint),
-                            JValue::Int(0),
-                        ],
-                    )
-                    .unwrap();
-                }
-                Some(diff::PatchResult::Replace { node, parent }) => {
-                    env.call_method(
-                        &interface,
-                        "mOnHandle",
-                        "(JBII)V",
-                        &[
-                            JValue::Long(this),
-                            JValue::Byte(ChangeType::Replace as jbyte),
-                            JValue::Int(node.as_u32() as jint),
-                            JValue::Int(parent.as_u32() as jint),
-                        ],
-                    )
-                    .unwrap();
-                }
+    if this.is_null() || other.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Document::merge called with null pointer",
+        )
+        .unwrap();
+
+        return;
+    }
```

Comment:
Here's a suggestion for a more specific error message

```suggestion
    let this_null = this.is_null();
    let other_null = other.is_null();
    
    if this_null() || other_null() {
        let message = String::new("Document::merged called with ");
        
        if this_null {
           message.push_str("null this pointer");
          
           if other_null {
              message.push_str(" and ");
            }
        }
        
        if other_null {
            message.push_str("null other pointer.");
        }
        
        env.throw_new(
            "java/lang/NullPointerException",
            message,
        )
        .unwrap();

        return;
    }
```